clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/CloneKubernetesAtomicOperation.groovy

```diff
@@ -0,0 +1,109 @@
+
+class CloneKubernetesAtomicOperation implements AtomicOperation<Void> {
+  private static final String BASE_PHASE = "CLONE_SERVER_GROUP"
+
+  @Autowired
+  KubernetesUtil kubernetesUtil
+
+  CloneKubernetesAtomicOperation(CloneKubernetesAtomicOperationDescription description) {
+    this.description = description
+  }
+
+  private static Task getTask() {
+    TaskRepository.threadLocalTask.get()
+  }
+
+  CloneKubernetesAtomicOperationDescription description
+
+/*
+ * curl -X POST -H "Content-Type: application/json" -d  '[ { "cloneServerGroup": { "source": { "serverGroupName": "k8s-test-v000" }, "credentials":  "my-k8s-account" } } ]' localhost:7002/kubernetes/ops
+ * curl -X POST -H "Content-Type: application/json" -d  '[ { "cloneServerGroup": { "stack": "prod", "freeFormDetails": "mdservice", "targetSize": "4", "source": { "serverGroupName": "k8s-test-v000" }, "credentials":  "my-k8s-account" } } ]' localhost:7002/kubernetes/ops
+*/
+  // TODO: should return DeploymentResult object
+  @Override
+  Void operate(List priorOutputs) {
+    CloneKubernetesAtomicOperationDescription newDescription = cloneAndOverrideDescription()
+
+    task.updateStatus BASE_PHASE, "Initializing copy of server group for " +
+      "${description.application}-${description.stack}..."
+
+    new DeployKubernetesAtomicOperation(newDescription).operate(priorOutputs)
+
+    task.updateStatus BASE_PHASE, "Finished copying server group for " +
+      "${description.application}-${description.stack}. "
+  }
+
+  // Made public for testability, not intended to be called directly
+  CloneKubernetesAtomicOperationDescription cloneAndOverrideDescription() {
+    CloneKubernetesAtomicOperationDescription newDescription = description.clone()
+
+    task.updateStatus BASE_PHASE, "Reading ancestor server group $description.source.serverGroupName..."
+
+    ReplicationController ancestorServerGroup = kubernetesUtil.getReplicationController(description.kubernetesCredentials, description.source.serverGroupName)
+
+    if (!ancestorServerGroup) {
+      throw IllegalArgumentException("Source server group $description.source.serverGroupName does not exist")
+    }
+
+    def ancestorNames = Names.parseName(description.source.serverGroupName)
+
+    // Build description object from ancestor, override any values that were specified on the clone call
+    newDescription.application = description.application ?: ancestorNames.app
+    newDescription.stack = description.stack ?: ancestorNames.stack
+    newDescription.freeFormDetails = description.freeFormDetails ?: ancestorNames.detail
+    newDescription.targetSize = description.targetSize ?: ancestorServerGroup.spec?.replicas
+    newDescription.loadBalancers = []
+    newDescription.securityGroups = []
+    ancestorServerGroup.spec?.template?.metadata?.labels?.each { key, val ->
+      if (key.startsWith("load-balancer-")) {
+        newDescription.loadBalancers.push(key.substring(14, key.length()))
```

Comment:
14 here and 15 below are magic numbers. 

1. Make them constants
1. When doing so, make it clear where it comes from, i.e.:
  ```
  public static final String LOAD_BALANCER_PREFIX_LENGTH = "load-balancer-".length()
  ```